stock_inventory/models/stock_quant.py

```diff
@@ -36,9 +39,32 @@ def _apply_inventory(self):
                 raise ValueError(_("No move lines have been created"))
             move = moves[len(moves) - 1]
             adjustment.stock_move_ids |= move
-            move.inventory_adjustment_id = adjustment
+            reference = move.reference
+            if adjustment.name and move.reference:
+                reference = adjustment.name + ": " + move.reference
+            elif adjustment.name:
+                reference = adjustment.name
+            move.write(
+                {
+                    "inventory_adjustment_id": adjustment.id,
+                    "reference": reference,
+                }
+            )
             rec.to_do = False
+        if adjustment and self.env.company.stock_inventory_auto_complete:
+            adjustment.action_auto_state_to_done()
         return res
 
     def _get_inventory_fields_write(self):
         return super()._get_inventory_fields_write() + ["to_do"]
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+        if self.env.context.get(
+            "active_model", False
+        ) == "stock.inventory" and self.env.context.get("active_id", False):
+            self.env["stock.inventory"].browse(
+                self.env.context.get("active_id")
+            ).refresh_stock_quant_ids()
+        return res
```

Comment:
Can't we simply add context default values and add related fields in the tree view? Like I did in the previous PR. Also you should use model_create_multi.

Reply:
Hi, I've changed the model_create_multi, however I don't understand exactly what you refer by context default values. You mean creating the quant with default values automatically? 

If you agree, we could merge this PR with the alignment changes and you can create a new PR with your changes added correctly. If not please tell me exaclty what do you mean please. Thanks!

Reply:
@DavidJForgeFlow I think this is what it is about:

![image](https://github.com/OCA/stock-logistics-warehouse/assets/19529533/c3fb48f4-dd35-4709-9723-ec03f36a8331)

Reply:
I've added the default_to_do as stock_inventory_ids is no longer in quants. I think we need to review if this fields is still needed, IMO it's not longer needed.

Reply:
I'm not a fan of this implementation. I wasn't a fan of what I did based on your previous algorithm but it was cleaner in my opinion. I should take time to analyze this and also include the rest of my PR, because here the merge is incomplete so I have things to to that was not planned.

Reply:
@DavidJForgeFlow Based on this response I think it is best to split this PR, between the strict fwport from v15 and the adjustments done by @benwillig in which we could continue this discussion.

I would like to have v15 and v16 aligned to unlock the migration of v16 cycle count.